test/intl402/Intl/getCanonicalLocales/duplicates.js

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 Mozilla Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: sec-intl.getcanonicallocales
+description: Tests the getCanonicalLocales function for duplicate locales scenario.
+info: |
+  8.2.1 Intl.getCanonicalLocales (locales)
+  1. Let ll be ? CanonicalizeLocaleList(locales).
+  2. Return CreateArrayFromList(ll).
+includes: [compareArray.js]
+---*/
+
+assert(compareArray(
+  Intl.getCanonicalLocales(
+    ['ab-cd', 'ff', 'de-rt', 'ab-Cd']), ['ab-CD', 'ff', 'de-RT']));
+
+var locales = Intl.getCanonicalLocales(["en-US", "en-US"]);
+assert(compareArray(locales, ['en-US']));
```

Comment:
When I suggested using compareArray I forgot to mention it does not make clear when it fails. In cases we have 2 or more assertions in the same file, is good to provide an assertion message so the implementor knows where it's failing. This can be made as:

`assert(compareArray(locales, ['en-US']), 'en-US');`

This way makes it easy to map what is failing.